compiler-core/src/erlang/tests/snapshots/gleam_core__erlang__tests__bit_arrays__bit_array.snap

```diff
@@ -20,7 +20,7 @@ pub fn main() {
 
 -export([main/0]).
 
--file("/root/project/test/my/mod.gleam", 1).
+-file("my/mod/project/test/my/mod.gleam", 1).
```

Comment:
Why do these start with `my/mod` now? Shouldn't they be `project/...`

Reply:
Sorry, I'm confused. Why do we use the module name as a prefix? That results in an invalid path to a file that doesn't exist?

Reply:
@lpil No worries. Here is how I understood this issue:
If you look at the #4141, here is actual `-file` directory being generated before the fix:
-file("/Users/USERNAME/SOME/PATH/some_lib_or_app/src/some_lib_or_app.gleam", 26).
And the expectation is:
-file("some_lib_or_app/src/some_lib_or_app.gleam", 26), which the fix is about. 
In our pipeline, I understood that the project getting parsed as `/Users/USERNAME/SOME/PATH`, the module name as `some_lib_or_app`.  Thus the fix.  Let me know if you think there is better way to fix this issue.

Reply:
I'm not sure I understand, so I'll give an example of what the behaviour should be.

Current working directory: `/root/project`
Source file location: `/root/project/src/wibble.gleam`
`-file` value: `src/wibble.gleam`

The tests don't seem to match this from what I can see, why is that?

Reply:
Ah, if that is what the expected behaviour, then the fix would be much simpler and I could update the code changes and tests to reflect that.  So to confirm:
Currently (main branch) generates fqn for the generated erlang -file directive like this:
`/root/project/src/wibble.gleam` given the cwd of `/root/project`, which is the bug.
The correct behaviour is:
the file directive should be `src/wibble.gleam`. 

Please confirm and will take care of this soon.